File: src/main/java/com/mysawit/payroll/MysawitPayrollServiceApplication.java
package com.mysawit.payroll;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.ConfigurableApplicationContext;

@SpringBootApplication
public class MysawitPayrollServiceApplication {
    public static void main(String[] args) {
        ConfigurableApplicationContext context = SpringApplication.run(MysawitPayrollServiceApplication.class, args);
        if (context.getEnvironment().getProperty("app.test.close-context", Boolean.class, false)) {
            context.close();
        }
    }
}

File: src/main/kotlin/com/mysawit/payroll/MysawitPayrollServiceApplication.kt
package com.mysawit.payroll

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication

@SpringBootApplication
class MysawitPayrollServiceApplication

fun main(args: Array<String>) {
    val context = runApplication<MysawitPayrollServiceApplication>(*args)
    if (context.environment.getProperty("app.test.close-context", Boolean::class.java, false)) {
        context.close()
    }
}
